oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,115 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+)
+
+type CloudConnectionDetails struct {
+	AccountId  *string `json:"account_id"`
+	ExternalId *string `json:"external_id"`
+	RoleArn    *string `json:"role_arn"`
+}
+
+type CloudConnection struct {
+	DeletedAt              *string                 `json:"deleted_at,omitempty"`
+	ID                     *string                 `json:"id,omitempty"`
+	Name                   *string                 `json:"name"`
+	Provider               *string                 `json:"provider"`
+	CloudConnectionDetails *CloudConnectionDetails `json:"cloud_connection_details"`
+}
+
+func (c CloudConnection) Exists() bool {
+	return utils.IsNonEmpty(c.ID) && utils.IsBlank(c.DeletedAt)
+}
+
+func (c CloudConnection) ToResourceMap() map[string]any {
+	m := make(map[string]any, 2)
+
+	if c.Name != nil {
+		m[attributes.Name] = *c.Name
+	}
+	if c.ID != nil {
+		m[attributes.ID] = *c.ID
+	}
+	if c.DeletedAt != nil {
+		m[attributes.DeletedAt] = *c.DeletedAt
+	}
+	m[attributes.CloudConnectionDetails] = c.CloudConnectionDetails
+
+	return m
+}
+
+func (c OktaPAMClient) GetCloudConnection(ctx context.Context, name string, allowDeleted bool) (*CloudConnection, error) {
```

Comment:
I'm not going to block this PR, but we are trying to add more to the `OktaPAMClient` and instead use the SDK client.  We haven't moved things over to the new SDK client yet, so I am guessing you saw code using the `OktaPAMClient` and based your changes off of that.  

If you have the time, please consider using the SDK client.

Reply:
+1